Blahaj/Assets/Scripts/PlayerStats.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerStats : MonoBehaviour
+{
+
+    #region Fields
+
+    private List<string> skillsList = new List<string>(); // can use an abstract skills class as well
+    [SerializeField] private int health;
+    [SerializeField] private int attackDamage; // float?
+    [SerializeField] private float attackSpeed; 
+    [SerializeField] private float movementSpeed;
+    [SerializeField] private int maxHealth;
+    [SerializeField] private int minAttackDamage;
+    [SerializeField] private int maxAttackDamage;
+    [SerializeField] private float minAttackSpeed;
+    [SerializeField] private float maxAttackSpeed;
+    [SerializeField] private float minMovementSpeed;
+    [SerializeField] private float maxMovementSpeed;
+    
+    private Dictionary<string, int> orbs = new Dictionary<string, int>(){
+        {"Red", 0},
+        {"Yellow", 0},
+        {"Purple", 0}
+    };
+    // port movement speed over, have some way for playerMovement to access this value
+
+    public delegate void IntStatsChange(int changeAmount);
+    public static IntStatsChange IncreaseAttackDamageEvent;
+    public static IntStatsChange DecreaseAttackDamageEvent;
+    public static IntStatsChange RestoreHealthEvent;
+    public static IntStatsChange DamageEvent;
+
+    public delegate void FloatStatsChange(float changeAmount);
+    public static FloatStatsChange IncreaseAttackSpeedEvent;
+    public static FloatStatsChange DecreaseAttackSpeedEvent;
+    public static FloatStatsChange IncreaseMovementSpeedEvent;
+    public static FloatStatsChange DecreaseMovementSpeedEvent;
+
+    public delegate float FloatGetter();
+    public static FloatGetter MovementSpeedGetter;
+
+    public delegate void OrbsChange(int red, int yellow, int purple);
+    public static OrbsChange GainOrbsEvent;
+    public static OrbsChange ConsumeOrbsEvent;
+
+    /*
+    public delegate void SkillGain(string skill);
+    public static SkillGain GainSkillEvent;
+
+    public delegate string SkillGetter(int index);
+    public static SkillGetter RetrieveSkillEvent;
+    */
+    
+    private SpriteRenderer mySR;
+
+    #endregion
+    
+    #region Methods
+    
+    // Start is called before the first frame update
+    private void Start()
+    {
+        PlayerStats.DamageEvent += Damage;
+        PlayerStats.MovementSpeedGetter += GetMovementSpeed;
+        PlayerStats.IncreaseAttackDamageEvent += IncreaseAttackDamage;
+        PlayerStats.DecreaseAttackDamageEvent += DecreaseAttackDamage;
+        PlayerStats.RestoreHealthEvent += RestoreHealth;
+        PlayerStats.IncreaseAttackSpeedEvent += IncreaseAttackSpeed;
```

Comment:
increase and decrease can change to just "ChangeAttackSpeed" or something on that lines, accept negative or positive value